tasks/DroneFlightControlTask.cpp

```diff
@@ -45,26 +46,90 @@ bool DroneFlightControlTask::startHook()
         return false;
 
     mLastMission = Mission();
-
-    mAuthorityStatus = mVehicle->obtainCtrlAuthority(mFunctionTimeout);
-    ACK::getErrorCodeMessage(mAuthorityStatus, __func__);
     return true;
 }
 
-typedef DroneFlightControlTask::States TaskState;
-static TaskState djiStatusFlightToTaskState(uint8_t status)
+DroneFlightControlTask::States
+DroneFlightControlTask::runtimeStatesTransition(Telemetry::SDKInfo control_status)
 {
-    switch (status)
+    DroneFlightControlTask::States current_state = state();
+    switch (current_state)
     {
-        case DJI::OSDK::VehicleStatus::FlightStatus::STOPED:
-            return TaskState::DJI_STOPPED;
-        case DJI::OSDK::VehicleStatus::FlightStatus::ON_GROUND:
-            return TaskState::DJI_ON_GROUND;
-        case DJI::OSDK::VehicleStatus::FlightStatus::IN_AIR:
-            return TaskState::DJI_IN_AIR;
+        case RUNNING:
+        {
+            return TELEMETRY;
+        }
+        case TELEMETRY:
+        {
+            if (_cmd_action.connected())
+            {
+                return CONTROLLING;
+            }
+            else
+            {
+                return TELEMETRY;
+            }
+        }
+        case CONTROLLING:
+        {
+            bool can_take_ctrl = canTakeControl(control_status);
+            if (can_take_ctrl)
+            {
+                return CONTROLLING;
+            }
+            else if (!_cmd_action.connected())
+            {
+                return TELEMETRY;
+            }
```

Comment:
This creates an inconsistency in the mapping between states and what is going on if we are in CONTROLLING and stop controlling:
- if the manual control does not take control we stay in controlling
- if it does, we transition to TELEMETRY

I think that it would be best to transition to TELEMETRY as soon as the command input is disconnected. It would be more helpful to understand what the component "thinks".